app/javascript/mastodon/components/status_content.js

```diff
@@ -46,10 +51,15 @@ export default class StatusContent extends React.PureComponent {
     }
   }
 
+  componentDidMount () {
+    this._updateStatusLinks();
+  }
+
   componentDidUpdate () {
     if (this.props.onHeightUpdate) {
       this.props.onHeightUpdate();
     }
+    this._updateStatusLinks();
```

Comment:
For all that it's worth, the reason I noticed the event handlers inconsistently getting set on CW placeholder links is because my instance has the instance-colors PR from a.weirder.earth merged in, and that changes the styling of mentions along with adding the event handlers.

Personally I think this way is better just for the sake of being future proof; I could see someone trying to implement something like editing statuses and running into this bug again.